xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -598,13 +598,15 @@ static final class StaticStrideScheduler {
       if (numWeightedChannels > 0) {
```

Comment:
The fact that the match works out doesn't change that it is somewhat confusing that it goes through this path for 1 weighted channel.  It would be much clearer if it was changed to (numWeightedChannels >= 2)

Reply:
The problem is reality; this is an emergent property. When there's one entry that doesn't have `0`, you need to either set that entry to zero, assign all other entries to the same value, or use custom weight-creation logic. The `numWeightedChannels > 0` path calculates the value of that entry, and duplicates it elsewhere. Setting that one entry to zero would require another pass through the entries. Basically, there's no way to combine `numWeightedChannels == 0` and `numWeightedChannels == 1` cases, unless we give them their own custom weight-creation logic. We could have three cases here instead of two... but that doesn't help us either.